### ISSUE TITLE
Feature Request: Translate to ES2015

### ISSUE THREAD

**Issue by marcobiedermann** [luruke/barba.js#39]
How about translating this awesome library to ES6? If you like to, I could help you out with it :)

**Comment by luruke** [comment]
Hi @marcobiedermann !
Actually I started writing the library in ES6, but then I moved to ES5.

What stopped me is that:
- I din't find a way to use ES6 Classes on ES5 (for instance, how do you extends?)
- There are not really benefits to write Barba.js in ES6, if not just for a "better" syntax
- I am sure that compiling using a transpiler (babel for example) the file size of the library will grow

But i'm open to discussion :)

Would be cool to do something like:

```javascript
class MyViews extends Barba.View {
...
}
```

```javascript
class Fade extends Barba.Transition {
...
}
```

**Comment by marcobiedermann** [comment]
1. ES2015 does supports extend, which gets transpiled to ES5 code
```
class View {
 … 
}

class MyView extends View {
 … 
}
```
[Example](https://babeljs.io/repl/#?evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-2&code=class%20View%20%7B%0A%20%20%0A%7D%0A%0Aclass%20MyView%20extends%20View%20%7B%0A%20%20%0A%7D)

2. Yes, you are right, most ES2015 features are syntactic sugar. In my personal opinion, I think it is easier to read and less code. Modern browsers, which support ES2015 (and they will grow in the future) benefit from that.

3. Yes, using a transpiler will grow the overall code, but when used with oder libraries which support ES2015, this will be a minor impact.

I am a huge fan of new features and standards and I like giving modern browsers the best way to handle code. ES2015 will be standard in a few years and so all code should be translated sooner or later to it.

This is just my opinion but I leave it up to your, if your are going to make this change :)
Still thanks for this awesome library 👍

**Comment by luruke** [comment]
Yes @marcobiedermann, I **completely** agree with you.
I use ES6 for almost my project at work I really like it and it without discussion it brings a lots of benefit.

Yes, the classes are correctly transpiled, but what is not clear to me, is how do you extend them using ES5? 😕 
I want people be able to use Barba.js on older ES version as well.

Another point (on which I am not really sure) is that using ES6 can decrease the number of people that can read / write the code and thus to contribute to the project.

To sum up, I find ES6 great for making projects / website, but it doesn't worth yet to be used for reusable libraries.

Again, this is an open discussion, I'm happy to know the pros and cons and then decide 😃

**Comment by terion-name** [comment]
about extention — I think the proper way to support both new and old syntax will be to write things in ES6 and add compatibility wrappers in ES5 bundle.

sort of:
```
// es6
class View {}

// for es5 bundle
var view = new View;
view.extend = function(){} // or use mixin, or HOC, whatever
```

So you provide in package es6 sources and a gulp-builded bundle for es5

**Comment by luruke** [comment]
Hey @terion-name ,
so you mean actually to serve two different build for ES6 and ES5?
Do you know if it's something easily feasible with a single npm package?

**Comment by terion-name** [comment]
for es6 you actually don't need to build I think. for what?
you provide es6 code in npm package with builded es5 bundle just placed inside.
```
import Barba from 'barba.js' // es6
var Barba = require('barba.js/lib'); // es5; lib, bundle, es5 or anything you like
```
I think this will work nice. But it's just my opinion, maybe some other users will have better ideas

**Comment by pgangwani** [comment]
@luruke : What about writing barbajs in Es6 and publish in ES5 to npm . 
Refer this : 
http://stackoverflow.com/questions/29738381/how-to-publish-a-module-written-in-es6-to-npm

Letme know if I am thinking different.

**Comment by luruke** [comment]
Hello @pgangwani ,
yes, that is clear to me.

The problem is

> Yes, the classes are correctly transpiled, but what is not clear to me, is how do you extend them using ES5? 😕 

I will try to think about a solution.
Cheers

**Comment by luruke** [comment]
Close this, we can continue on https://github.com/luruke/barba.js/issues/41
